doc/dev/developer_guide/index.rst

```diff
@@ -1586,6 +1586,94 @@ server list`` on the teuthology machine, but the target VM hostnames (e.g.
 ``target149202171058.teuthology``) are resolvable within the teuthology
 cluster.
 
+Runing Integration Tests using Teuthology
+=========================================
+
+Getting binaries
+----------------
+To run integration test using teuthology, you need to have Ceph binaries built
+for your branch. Following is how you can do it:
+
+#. Push your branch to `ceph-ci`_ repository. This would trigger the build
+   process.
+
+#. To confirm that the build process has initiated, spot your branch name
+   at `shaman`_. Little after the build process has initiated, the single
+   entry with your branch name would multiply, each new entry for a different
+   OS.
+
+#. Wait for building process to complete. Usually, it takes around 2-3 hours
+   depending on how busy the machines are.
+
+Triggering tests
+-----------------
+After building is complete are ready, proceed to trigger tests -
+
+#. Log in to the teuthology machine::
+
+       ssh user@teuthology.front.sepia.ceph.com
+
+   You would need access to sepia lab for this. If you don't have one, you can
+   request that.
+
+#. You would need to get teuthology commands installed. Run the first set of
+   commands in `Running Your First Test`_. After that, activate the virtual
+   environment where teuthology is installed.
+
+#. Run the ``teuthology-suite`` command::
+
+        teuthology-suite -v -m smithi -c wip-cephfs-shell-feature-x -s fs -p 50 --filter "cephfs-shell"
+
+   Following are the options used in above command with their meaning -
+        -v          verbose
+        -m          machine name
+        -c          branch name, the branch that was pushed on cep-ci
+        -s          test-suite name
+        -p          priority number; see `Testing Priority`_
+        --filter    filter tests in given suite that needs to run, the arg to
+                    filter should be the test you want to run
+
+#. Wait for the tests to run. ``teuthology-suite``'s prints a link to the
+   `pulpito`_ page created for your tests. Once the tests have finished
+   running, it's more convenient to download and then view the log than
+   viewing it in a internet browser since these logs can easily be upto size
+   of 1 GB.
```

Comment:
May be just mention how to locate the `teuthology.log` and teuthology-archives in teuthology.front.sepia.ceph.com?

Reply:
Agreed, I'll elaborate on that.

Reply:
Added.